llvm/lib/Transforms/Utils/CodeLayout.cpp

```diff
@@ -1027,7 +1027,7 @@ class CDSortImpl {
 
     LLVM_DEBUG(dbgs() << "Cache-directed function sorting reduced the number"
                       << " of chains from " << NumNodes << " to "
-                      << HotChains.size() << "\n");
+                      << NumHotChains << "\n");
```

Comment:
One can move the debug statement to `concatChains()` where we have access to the number of remaining chains, `SortedChains.size()`. So this variable, NumHotChains, won't be needed